src/components/HappinessTable.tsx

```diff
@@ -0,0 +1,177 @@
+import { useEffect, useState } from "react";
+import { useQuery } from "react-query";
+import { QueryKeys } from "../constants";
+import { useApi } from "../contexts/ApiProvider";
+import { Group } from "../data/models/Group";
+import { Happiness } from "../data/models/Happiness";
+import { floatToColor, formatDate, parseYYYYmmddFormat } from "../utils";
+import HappinessViewerModal from "./modals/HappinessViewerModal";
+/**
+ *
+ * @param startDate a string in the YYYY-MM-DD format representing the start of
+ * the happiness date range
+ * @param endDate a string in the YYYY-MM-DD format representing the end of
+ * the happiness date range (defaults to today if unspecified)
+ * @param groupId the associated group ID for the happiness table
+ * @returns
+ */
+export default function HappinessTable({
+  group,
+  startDate,
+  endDate = formatDate(new Date()),
+}: {
+  group: Group;
+  startDate: string;
+  endDate?: string;
+}) {
+  const { api } = useApi();
+  const [selectedHappiness, setSelectedHappiness] = useState<
+    Happiness | undefined
+  >(undefined);
+  // open modal when user selects new happiness
+  useEffect(() => {
+    if (selectedHappiness) {
+      window.HSOverlay.open(document.querySelector("#view-happiness"));
```

Comment:
should we also mark the post as read when opening the modal or should that only be for the feed

Reply:
yes, just did that